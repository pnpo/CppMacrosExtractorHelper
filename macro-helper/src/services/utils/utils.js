import _ from 'lodash';

class Utils {

    sort = (arr, name, order) => {

        order = order ? order : 'asc';

        return _.orderBy(arr, [s => {

            if (typeof s[name] == 'string') {
            
                return s[name].toLowerCase();
            }

            return s[name];            
            
        }], [order]);
    }
}

export default new Utils();