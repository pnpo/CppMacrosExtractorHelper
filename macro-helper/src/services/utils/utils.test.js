import utils from './utils';

describe('utils test suite', () => {

    let arr = [];

    expect.addSnapshotSerializer({
        test: (val) => {

            if (val != undefined) {
                
                return ;
            }
        },
        print: (val) => `${val.name}`
    });

    beforeEach(() => {

        arr = [{ name: 'lenon' }, { name: 'abramson' }, { name: 'zidan' }, { name: 'marley' }];    
    });

    it('sort string empty array', () => {

        const array = undefined;

        const result = utils.sort(array, 'name');

        expect(result).toMatchSnapshot();
    });

    it('sort string empty array - not sorted', () => {

        arr[1].name = undefined;

        const result = utils.sort([], 'name');

        expect(result).toMatchSnapshot();
    });

    it('sort string array without property name', () => {

        const result = utils.sort(arr);

        expect(result).toMatchSnapshot();
    });

    it('sort string array with undefined item', () => {

        arr[1].name = undefined;

        const result = utils.sort(arr, 'name');

        expect(result).toMatchSnapshot();
    });

    it('sort string array with empty item', () => {

        arr[1].name = '';

        const result = utils.sort(arr, 'name');

        expect(result).toMatchSnapshot();
    });

    it('sort string array with integer item', () => {

        arr[1].name = 3;

        const result = utils.sort(arr, 'name');

        expect(result).toMatchSnapshot();
    });
    
    it('sort string array defualt asc when order not defined', () => {

        const result = utils.sort(arr, 'name');

        expect(result).toMatchSnapshot();
    });

    it('sort string array asc', () => {

        const result = utils.sort(arr, 'name', 'asc');

        expect(result).toMatchSnapshot();
    });

    it('sort string array desc', () => {

        const result = utils.sort(arr, 'name', 'desc');

        expect(result).toMatchSnapshot();
    });

    it('sort string array asc case sensitive', () => {

        arr[0].name = 'LENON';
        
        const result = utils.sort(arr, 'name', 'asc');

        expect(result).toMatchSnapshot();
    });
});
