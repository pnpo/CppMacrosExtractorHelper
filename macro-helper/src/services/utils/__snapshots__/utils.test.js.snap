// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils test suite sort string array asc 1`] = `
Array [
  Object {
    "name": "abramson",
  },
  Object {
    "name": "lenon",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
]
`;

exports[`utils test suite sort string array asc case sensitive 1`] = `
Array [
  Object {
    "name": "abramson",
  },
  Object {
    "name": "LENON",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
]
`;

exports[`utils test suite sort string array defualt asc when order not defined 1`] = `
Array [
  Object {
    "name": "abramson",
  },
  Object {
    "name": "lenon",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
]
`;

exports[`utils test suite sort string array desc 1`] = `
Array [
  Object {
    "name": "zidan",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "lenon",
  },
  Object {
    "name": "abramson",
  },
]
`;

exports[`utils test suite sort string array with empty item 1`] = `
Array [
  Object {
    "name": "",
  },
  Object {
    "name": "lenon",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
]
`;

exports[`utils test suite sort string array with integer item 1`] = `
Array [
  Object {
    "name": "lenon",
  },
  Object {
    "name": 3,
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
]
`;

exports[`utils test suite sort string array with undefined item 1`] = `
Array [
  Object {
    "name": "lenon",
  },
  Object {
    "name": "marley",
  },
  Object {
    "name": "zidan",
  },
  Object {
    "name": undefined,
  },
]
`;

exports[`utils test suite sort string array without property name 1`] = `
Array [
  Object {
    "name": "lenon",
  },
  Object {
    "name": "abramson",
  },
  Object {
    "name": "zidan",
  },
  Object {
    "name": "marley",
  },
]
`;

exports[`utils test suite sort string empty array - not sorted 1`] = `Array []`;

exports[`utils test suite sort string empty array 1`] = `Array []`;
